src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/DebugProfileEnumValuesGenerator.cs

```diff
@@ -28,17 +30,32 @@ internal class DebugProfileEnumValuesGenerator : IDynamicEnumValuesGenerator
         /// </summary>
         internal DebugProfileEnumValuesGenerator(ILaunchSettingsProvider profileProvider)
         {
-            listedValues = new AsyncLazy<ICollection<IEnumValue>>(delegate 
+            this.listedValues = new AsyncLazy<ICollection<IEnumValue>>(delegate
             {
                 var curSnapshot = profileProvider.CurrentSnapshot;
-                if (curSnapshot != null )
+                if (curSnapshot != null)
                 {
                     return Task.FromResult(GetEnumeratorEnumValues(curSnapshot));
                 }
 
                 ICollection<IEnumValue> emptyCollection = new List<IEnumValue>();
                 return Task.FromResult(emptyCollection);
-            });
+            }, TryGetJoinableTaskFactory());
+        }
+
+        private static JoinableTaskFactory TryGetJoinableTaskFactory()
+        {
+            JoinableTaskFactory factory = null;
+            try
+            {
+                factory = ThreadHelper.JoinableTaskFactory;
+            }
+            catch (System.NullReferenceException)
```

Comment:
I do not like the approach of catching the NullReference. Instead, can we please pull in IProjectThreadingService and its factory, which we should be anyway.